src/webui/api/torrentscontroller.cpp

```diff
@@ -559,12 +559,43 @@ void TorrentsController::filePrioAction()
     checkParams({"hash", "id", "priority"});
 
     const QString hash = params()["hash"];
-    int fileID = params()["id"].toInt();
-    int priority = params()["priority"].toInt();
+    bool ok = false;
+    const int priority = params()["priority"].toInt(&ok);
+    if (!ok)
+        throw APIError(APIErrorType::BadParams, tr("Priority must be an integer"));
+
+    if (!BitTorrent::FilePriority::isValidPriority(priority))
+        throw APIError(APIErrorType::BadParams, tr("Priority is not valid"));
+
     BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (!torrent)
+        throw APIError(APIErrorType::NotFound);
+    if (!torrent->hasMetadata())
+        throw APIError(APIErrorType::Conflict, tr("Torrent's metadata has not yet downloaded"));
+
+    const int filesCount = torrent->filesCount();
+    QList<int> fileIds;
+    for (const QString &fileID : params()["id"].split('|')) {
+        const int id = fileID.toInt(&ok);
+        if (!ok)
+            throw APIError(APIErrorType::BadParams, tr("File IDs must be integers"));
+        if ((id < 0) || (id >= filesCount))
+            throw APIError(APIErrorType::NotFound, tr("File ID is not valid"));
+        fileIds << id;
+    }
+
+    QVector<int> priorities = torrent->filePriorities();
+    bool priorityChanged = false;
+
+    for (const int &id : fileIds) {
```

Comment:
Why not to merge this loop with the one above?

Reply:
It made sense before, but can be refactored now that we use `prioritizeFiles()`.